File: code_forces/96A/A.java
import java.util.*;
import java.io.*;

public class A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String players = sc.nextLine();

        char[] line = players.toCharArray();

        int ans1 = 0;
        int ans2 = 0;
        for(char x: line) {
            if(ans1 == 7 || ans2 == 7) {
                break;
            }
            if(x == '1') {
                ans1++;
                ans2 = 0;
            } else {
                ans2++;
                ans1 = 0;
            }
        }

        if(ans1 == 7 || ans2 == 7) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }
}

File: code_forces/96A/a.py
import sys

def main():
    players = sys.stdin.readline()

    ans1 = 0; ans2 = 0
    for i in players:
        if ans1 == 7 or ans2 == 7:
            break
        else:
            if i == '1':
                ans1 += 1
                ans2 = 0
            elif i == '0':
                ans2 += 1
                ans1 = 0

    if ans1 == 7 or ans2 == 7:
        sys.stdout.write('YES')
    else:
        sys.stdout.write('NO')


if __name__ == '__main__':
    main()